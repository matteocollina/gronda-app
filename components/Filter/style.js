import { StyleSheet } from "react-native";
import { PRIMARY_COLOR } from "../../assets/consts";

export default style = StyleSheet.create({
    container: { overflow: 'hidden',
    borderRadius: 20,
    borderWidth: 1,
    borderColor: 'black',
    padding: 10, },
})