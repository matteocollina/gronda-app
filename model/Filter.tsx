export default class FilterData{
  private _id: number;
  private _title: string;

  constructor(id: number, title: string) {
    this._id = id;
    this._title = title;
  }

  get id(): number {
    return this._id;
  }
  get title(): string {
    return this._title;
  }
}
