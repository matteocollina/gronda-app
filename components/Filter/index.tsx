import {FC} from 'react';
import React from 'react';
import { GestureResponderEvent, StyleProp, Text, TouchableOpacity, View, ViewStyle } from 'react-native';
import style from './style';

interface IFilter {
  isActive?: boolean;
  title: string;
  containerStyle?: StyleProp<ViewStyle>;
  onPress: (event: GestureResponderEvent) => void;
  testId?: string;
}

const Filter: FC<IFilter> = ({isActive, title, containerStyle, onPress, testId}: IFilter) => {
  return (
    <TouchableOpacity onPress={onPress} 
    testID={testId}
    style={[{
        backgroundColor: isActive ? 'black' : 'white',
        },style.container,containerStyle]}>
        <View >
        <Text
        style={{fontWeight: 'bold',
        color: !isActive ? 'black' : 'white',}}
            >
            {title}
        </Text>
        </View>
    </TouchableOpacity>
  );
};
export default Filter;
