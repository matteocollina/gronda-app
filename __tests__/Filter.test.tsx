import React from 'react'
import '@testing-library/jest-dom'
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import Filter from "../components/Filter";
import renderer from 'react-test-renderer';

describe("Filter Component",()=>{
    test('funtion onPress is called only one time', async () => {
        const fn = jest.fn();
        render(<Filter title={"Jobs"} onPress={fn} />)
        fireEvent.click(screen.getByText('Jobs'))
        expect(fn).toBeCalledTimes(1)
      })

      test('renders correctly', async () => {
        const fn = jest.fn();
        const tree = renderer
            .create(<Filter title={"Jobs"} onPress={fn} testId={"Filter"}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
      })
})
