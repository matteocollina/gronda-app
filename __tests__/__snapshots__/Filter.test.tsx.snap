// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 20,
      "borderWidth": 1,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 10,
    }
  }
  testID="Filter"
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontWeight": "bold",
        }
      }
    >
      Jobs
    </Text>
  </View>
</View>
`;
